IPython/terminal/interactiveshell.py

```diff
@@ -317,6 +317,11 @@ def _displayhook_class_default(self):
         help="Allows to enable/disable the prompt toolkit history search"
     ).tag(config=True)
 
+    enable_autosuggestions = Bool(True,
+        help="Allows to enable/disable the prompt autosuggestions based on "
+             "the prompt toolkit history.",
+    ).tag(config=True)
+
```

Comment:
Just thinking at loud, do you think there might be other source of autosuggestions we could have latter ? Would it make sens to make that a string, so we can select from multiple sources later ?

Reply:
Yes, that is a possibility. I had a look at `prompt_toolkit` and they have the infrastructure in place but as that are of their code has not changed in the last two years it feels like that might not happen in the near future. 

Of course there are other autosuggestion sources under development (Kite, IntelliCode) but I guess that requires a much larger integration effort and then this configuration option has to change in any case.

Reply:
So to make this a bit more future-proof I changed this to a string. That was definitely a good suggestion. 

Additionally it can be changed in a running terminal which might be useful as well.